packages/flutter/lib/src/material/chip.dart

```diff
@@ -2795,3 +2811,88 @@ class _RenderChip extends RenderBox {
   @override
   bool hitTestSelf(Offset position) => deleteButtonRect.contains(position) || pressRect.contains(position);
 }
+
+class _LocationAwareInkRippleFactory extends InteractiveInkFeatureFactory {
+  const _LocationAwareInkRippleFactory(
+    this.hasDeleteButton,
+    this.chipContext,
+    this.deleteIconKey,
+  );
+
+  final bool hasDeleteButton;
+  final BuildContext chipContext;
+  final GlobalKey deleteIconKey;
+
+  @override
+  InteractiveInkFeature create({
+    MaterialInkController controller,
+    RenderBox referenceBox,
+    Offset position,
+    Color color,
+    TextDirection textDirection,
+    bool containedInkWell = false,
+    RectCallback rectCallback,
+    BorderRadius borderRadius,
+    ShapeBorder customBorder,
+    double radius,
+    VoidCallback onRemoved,
+  }) {
+
+    final bool tapIsOnDeleteIcon = _tapIsOnDeleteIcon(
+      hasDeleteButton: hasDeleteButton,
+      tapPosition: position,
+      chipSize: chipContext.size,
+      textDirection: textDirection,
+    );
+
+    final BuildContext splashContext = tapIsOnDeleteIcon
+        ? deleteIconKey.currentContext
+        : chipContext;
+
+    final InteractiveInkFeatureFactory splashFactory = Theme.of(splashContext).splashFactory;
+
+    if (tapIsOnDeleteIcon) {
+      final RenderBox currentBox = referenceBox;
+      referenceBox = deleteIconKey.currentContext.findRenderObject();
+      position = referenceBox.globalToLocal(currentBox.localToGlobal(position));
+      containedInkWell = false;
+    }
+
+    return splashFactory.create(
+      controller: controller,
+      referenceBox: referenceBox,
+      position: position,
+      color: color,
+      textDirection: textDirection,
+      containedInkWell: containedInkWell,
+      rectCallback: rectCallback,
+      borderRadius: borderRadius,
+      customBorder: customBorder,
+      radius: radius,
+      onRemoved: onRemoved,
+    );
+  }
+}
+
+bool _tapIsOnDeleteIcon({
```

Comment:
This function is called after the delete icon has been laid out. Why aren't we using the delete icon RenderBox's [hitTest](https://api.flutter.dev/flutter/rendering/RenderBox/hitTest.html) to decide if it has been hit?

Reply:
Thank you for your message!

This is the same logic used by the previous version. Since I am only fixing the "double ripple" issue in this PR, I will keep it unchanged here. I can definitely file a separate issue regarding this, and we can fix it in a separate PR.

Reply:
Sounds good; please file an issue. We should be assuming that the delete icon has been hit if the icon itself is hit per hitTest(), or - assuming that the icon is smaller than 48x48 - the tap occurs within a 48x48 area centered on the icon.

Reply:
Issue filed — Chip delete button tap target is always 1/3 the width of the whole chip. #43291